src/num/fft.c

```diff
@@ -205,26 +205,53 @@ struct fft_plan_s {
 
 static DEF_TYPEID(fft_plan_s);
 
+static char* fftw_wisdom_name(unsigned int N, bool backwards, unsigned int flags, const long dims[N]) {
+
+	char* tbpath = getenv("TOOLBOX_PATH");
+	if (NULL == tbpath)
+		return NULL;
+
+	char STR_PATH[] = "/save/fftw/";
+	char STR_N[]    = "N_";
+	char STR_BACK[] = "_BACKWARD_";
+	char STR_FLAG[] = "_FLAGS_";
+	char STR_DIMS[] = "_DIMS";
+
+	int space[] = {strlen(tbpath), strlen(STR_PATH), strlen(STR_N), floor(log10(N)) + 1, strlen(STR_BACK), 1, strlen(STR_FLAG), floor(log10(flags)) + 1, strlen(STR_DIMS)};
+	size_t total = 1;                         // Start from one for null terminator.
+	for (size_t idx = 0; idx < sizeof(space)/sizeof(int); idx ++)
+		total += space[idx];                    // Space for base path.
+	for (size_t idx = 0; idx < N; idx ++)
+		total += (2 + floor(log10(dims[idx]))); // Space for dimensions.
+	total += 5;                               // Space for extension.
+
+	char* name = calloc(total, sizeof(char));
+	sprintf(name, "%s%s%s%u%s%u%s%u%s", tbpath, STR_PATH, STR_N, N, STR_BACK, backwards, STR_FLAG, flags, STR_DIMS);
+
+	char tmp[64];                             // Safe assumption since largest int in a 64 bit system is less than 20 places in base 10.
+	for (size_t idx = 0; idx < N; idx++) {
+		sprintf(tmp, "_%lu", dims[idx]);
+		strcat(name, tmp);
+	}
+
+	sprintf(tmp, ".fftw");
+	strcat(name, tmp);
+	name[total - 1] = '\0';
+
+	return name;
+}
+
```

Comment:
Why are you writing your own complicated logic to compute the size of the buffer?
Please use snprintf with a zero length string. It will give you the size needed (you need to add one for final null byte) or even better asprintf which will allocate the string for  you (not sure this exists on OS X).

Reply:
ok much better but snprintf gives you the size without the final zero. so just sum it up (without subtracting one) add one at the end.

Reply:
Whoops, you're right. I mis-counted when I was figuring out convention.